### PR TITLE
fix: undefined response from get-github-info call

### LINKED ISSUE #509: undefined result on get-github-info call
## Affected Packages
get-github-info

## Problem
When configuring changeset, if you pass in the repo name of the wrong format e.g `https://github.com/user/repo` instead of just `user/repo` it will cause the call to the Github API to return `undefined` and therefore causing `Cannot read property 'author' of undefined`. This error message can be misleading/confusing and so I think we need to handle that and show a nicer message.

There is an actual github action with that error of the repo that I'm working on - https://github.com/sajari/sdk-react/runs/1668436313?check_suite_focus=true

## Proposed solution
The solution is to add a more specific/stricter check for `request.repo` field to make sure that it follows the format (`userOrOrg/repoName`)

### PR DESCRIPTION
**Pull request by tuanddd**
Fixes #509

### CODE REVIEW

File: packages/get-github-info/src/index.ts [hunk 1]
```diff
@@ -2,6 +2,8 @@
 import fetch from "node-fetch";
 import DataLoader from "dataloader";
 
+const validRepoNameRegex = /^[A-Za-z0-9_.-]+\/[A-Za-z0-9_.-]+$/m;
```
Comment (Andarist) [inline packages/get-github-info/src/index.ts:5]: Do you have any source which described what repo names are considered as valid by GitHub? Are dots allowed? Why `m` modifier has been used?

Comment (tuanddd) [reply packages/get-github-info/src/index.ts:5]: The Regex was taken from [this answer on SO](https://stackoverflow.com/a/59082561)

And yes a little oversight on my end, the `m` flag is redundant, will remove it, thanks for catching that 👍

Comment (Andarist) [inline packages/get-github-info/src/index.ts:5]: This should be the equivalent
```suggestion
const validRepoNameRegex = /^[\w.-]+\/[\w.-]+$/;
```

File: packages/get-github-info/src/index.ts [hunk 2]
```diff
@@ -121,7 +123,7 @@ export async function getInfo(
     throw new Error("Please pass a commit SHA to getInfo");
   }
 
-  if (!request.repo) {
+  if (!request.repo || !validRepoNameRegex.test(request.repo)) {
```
Comment (Andarist) [inline packages/get-github-info/src/index.ts:126]: could you move the second check to a separate if and throw an error that would contain `validRepoNameRegex.source`? That way users should be able to get a little bit more information as to why the validation has failed

**Comment by tuanddd** [comment]
@Andarist I've pushed the changes 🙂

**Comment by tuanddd** [comment]
@Andarist What is blocking this?

**Review by Andarist — APPROVED** [review]

**Comment by Andarist** [comment]
> @Andarist What is blocking this?

My life :P

File: packages/get-github-info/src/index.ts (hunk 1 above)
Comment (chj-damon) [reply packages/get-github-info/src/index.ts]: why don't we support @ ? like '@ant-design/xxx'

**Pull request merged**